In [1]:
import pandas as pd

In [2]:
hotel_data_path = '../data/raw/hotel_booking.csv'

In [3]:
raw_df = pd.read_csv(hotel_data_path)

In [5]:
raw_df.columns

Index(['hotel', 'is_canceled', 'lead_time', 'arrival_date_year',
       'arrival_date_month', 'arrival_date_week_number',
       'arrival_date_day_of_month', 'stays_in_weekend_nights',
       'stays_in_week_nights', 'adults', 'children', 'babies', 'meal',
       'country', 'market_segment', 'distribution_channel',
       'is_repeated_guest', 'previous_cancellations',
       'previous_bookings_not_canceled', 'reserved_room_type',
       'assigned_room_type', 'booking_changes', 'deposit_type', 'agent',
       'company', 'days_in_waiting_list', 'customer_type', 'adr',
       'required_car_parking_spaces', 'total_of_special_requests',
       'reservation_status', 'reservation_status_date', 'name', 'email',
       'phone-number', 'credit_card'],
      dtype='object')

In [6]:
raw_df.shape[0]*.3

35817.0

In [43]:
raw_df['babies'].value_counts()

babies
0     118473
1        900
2         15
10         1
9          1
Name: count, dtype: int64

In [ ]:
raw_df[['babies', 'is_canceled']].groupby('babies').mean()

,is_canceled
babies,
0,0.371874
1,0.183333
2,0.133333
9,0.000000
10,0.000000


In [46]:
raw_df.isnull().sum()

hotel                                  0
is_canceled                            0
lead_time                              0
arrival_date_year                      0
arrival_date_month                     0
arrival_date_week_number               0
arrival_date_day_of_month              0
stays_in_weekend_nights                0
stays_in_week_nights                   0
adults                                 0
children                               4
babies                                 0
meal                                   0
country                              488
market_segment                         0
distribution_channel                   0
is_repeated_guest                      0
previous_cancellations                 0
previous_bookings_not_canceled         0
reserved_room_type                     0
assigned_room_type                     0
booking_changes                        0
deposit_type                           0
agent                              16340
company         

## missing values
threshold = 30% missing = 35,817</br>
stays_in_weekend_nights 0 = 51,998</br>
children: 0 = 110,796</br>
babies: 0 = 118,473</br>
is_repeated_gust: 0 = 115,580</br>
previous_cancellations: 0 = 112,906</br>
previous_bookings_not_cancelled: 0 = 115,770<br>
booking_changes: 0 = 101,314</br>
days_in_waiting_list: 0 = 115,692</br>
required_car_parking_spaces: 0 = 111,974</br>
total_of_special_requests: 0 = 70,318</br>



## attributes to remove + reason
reservation_status: already captured, encoded in is_canceled where No-Show is also encoded as a cancelation.

reservation_status_date: already encoded in columns arrival_date_year, arrival_date_month, arrival_date_week_number and arrival_date_day_of_the_month

name, email, phonen number, and credit card: do not need these attributes for a regression model.

company and agent: Company has a large number of missing values; agent and company are not relevant to the analysis and can be dropped. 